### PR TITLE
enh(platformstatus) insert topology in new table

### PR DESCRIPTION
**Pull request by loiclau**
## Description

Add existing nagios_server & remote_servers entries into the new topology table

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/install/php/Update-20.10.0.post.php [hunk 1]
```diff
@@ -143,3 +143,56 @@
...
+
+if (!empty($map)) {
+    $mapIp = parse_url(gethostbyname($map))["host"];
+    $stmt = $pearDB->prepare(
+        "INSERT INTO `platform_topology` (`address`, `name`, `type`, `parent_id`, `server_id`)
+            VALUES (:mapAddress, :name, 'map', :id, NULL)"
+    );
+    $name = 'map_' . $mapIp;
+    $stmt->bindValue(':mapAddress', $mapIp, \PDO::PARAM_STR);
+    $stmt->bindValue(':name', $name, \PDOap::PARAM_STR);
```
Comment (loiclau) [inline www/install/php/Update-20.10.0.post.php:163]:
```suggestion
    $stmt->bindValue(':name', $name, \PDO::PARAM_STR);
```

**Review by jeremyjaouen — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

File: www/install/php/Update-20.10.0.post.php [hunk 2]
```diff
@@ -143,3 +143,56 @@
...
+
+while ($row = $childStmt->fetch()) {
+    //check remote/poller
+    $parent = $parentId;
+    $serverType = 'poller';
+    if ($row['remote_id']) {
+        $parent = $row['remote_id'];
+    }
+
+    $remoteServerQuery = $pearDB->query("SELECT `id` FROM remote_servers WHERE ip = " . $row['ns_ip_address']);
```
Comment (kduret) [inline www/install/php/Update-20.10.0.post.php:181]:
```suggestion
    $remoteServerQuery = $pearDB->query("SELECT `id` FROM remote_servers WHERE ip = '" . $row['ns_ip_address'] . "'");
```

File: www/install/php/Update-20.10.0.post.php [hunk 3]
```diff
@@ -143,3 +143,56 @@
...
+    }
+
+    $stmt = $pearDB->prepare(
+        "INSERT INTO `platform_topology` (`address`, `name`, `type`, `parent_id`, `server_id`)
+         VALUES (:centralAddress, :name, :serverType, :parent, :id)"
+    );
+    $stmt->bindValue(':centralAddress', $row['ns_ip_address'], \PDO::PARAM_STR);
+    $stmt->bindValue(':name', $row['name'], \PDO::PARAM_STR);
+    $stmt->bindValue(':serverType', $serverType, \PDO::PARAM_STR);
+    $stmt->bindValue(':parent', $parent, \PDO::PARAM_INT);
```
Comment (kduret) [inline www/install/php/Update-20.10.0.post.php:195]:
```suggestion
    $stmt->bindValue(':parent', (int)$parent, \PDO::PARAM_INT);
```

**Review by kduret — DISMISSED** [review]

File: www/install/php/Update-20.10.0.post.php [hunk 4]
```diff
@@ -143,3 +143,56 @@
     );
     throw new \Exception($versionOfTheUpgrade . $errorMessage, (int)$e->getCode(), $e);
 }
+
+// get local server id
+$localStmt = $pearDB->query("SELECT `id` FROM nagios_server WHERE localhost = '1'");
```
Comment (sc979) [inline www/install/php/Update-20.10.0.post.php:148]: these queries should be added in the upper transaction commit / rollback block

**Review by sc979 — CHANGES_REQUESTED** [review]

**Review by sc979 — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: www/install/php/Update-20.10.0.post.php [hunk 5]
```diff
@@ -138,6 +138,63 @@
...
+        $errorMessage = "Unable to insert " . $serverType . ":" . $row['name'] . " in 'topology' table.";
+        $stmt = $pearDB->prepare(
+            "INSERT INTO `platform_topology` (`address`, `name`, `type`, `parent_id`, `server_id`)
+            VALUES (:centralAddress, :name, :serverType, :parent, :id)"
+        );
+        $stmt->bindValue(':centralAddress', $row['ns_ip_address'], \PDO::PARAM_STR);
+        $stmt->bindValue(':name', $row['name'], \PDO::PARAM_STR);
+        $stmt->bindValue(':serverType', $serverType, \PDO::PARAM_STR);
+        $stmt->bindValue(':parent', (int)$parent, \PDO::PARAM_INT);
+        $stmt->bindValue(':id', (int)$row['id'], \PDO::PARAM_INT);
```
Comment (jeremyjaouen) [inline www/install/php/Update-20.10.0.post.php:194]:
```suggestion
        $stmt->bindValue(':parent', (int) $parent, \PDO::PARAM_INT);
        $stmt->bindValue(':id', (int) $row['id'], \PDO::PARAM_INT);
```

**Review by jeremyjaouen — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**